Content.Server/Xenoarchaeology/Artifact/XenoArtifactSystem.ProcGen.cs

```diff
@@ -53,8 +53,14 @@ private List<XenoArchTriggerPrototype> CreateTriggerPool(Entity<XenoArtifactComp
             var triggerId = RobustRandom.Pick(weightsByTriggersLeft);
             weightsByTriggersLeft.Remove(triggerId);
             var trigger = PrototypeManager.Index<XenoArchTriggerPrototype>(triggerId);
-            if (_entityWhitelist.IsWhitelistFail(trigger.Whitelist, ent))
+            // DeltaV - start of TriggerBlacklist
+            if (
+                _entityWhitelist.IsWhitelistFail(trigger.Whitelist, ent)
+                || (trigger.TriggerBlacklist != null && triggerPool.Any(otherTrigger => trigger.TriggerBlacklist.Contains(otherTrigger.ID)))
+                || (triggerPool.Any(otherTrigger => otherTrigger.TriggerBlacklist != null && otherTrigger.TriggerBlacklist.Contains(triggerId)))
+            )
                 continue;
+            // DeltaV - end of TriggerBlacklist
 
```

Comment:
Remove extra parentheses, move comments so we're not marking the original line as part of the dV block
```suggestion
            var trigger = PrototypeManager.Index<XenoArchTriggerPrototype>(triggerId);
            if (
                _entityWhitelist.IsWhitelistFail(trigger.Whitelist, ent)
                || trigger.TriggerBlacklist != null && triggerPool.Any(otherTrigger => trigger.TriggerBlacklist.Contains(otherTrigger.ID))     // DeltaV - add blacklist of conflicting triggers
                || triggerPool.Any(otherTrigger => otherTrigger.TriggerBlacklist != null && otherTrigger.TriggerBlacklist.Contains(triggerId)) // DeltaV - add blacklist of conflicting triggers
            )
                continue;

```

Reply:
On past PRs I have been told to lean towards a wider comment net in cases like this (an if-statement condition), rather than dissecting units of code.

I gather that the reason is to make it less likely that the comment gets lost/missed when glancing over it, and also to reduce clutter. In reality, anyone doing merge conflicts (which is mainly why we add these comments) is also going to be considering this if-statement as a whole, and will have the upstream code side-by-side to compare.

So I'd prefer not to do this, since I'm receiving mixed signals.

If you really want me to, I could split this into a separate `if(new_conditions) continue;` statement, right after the original one.

Reply:
Alternatively, you can do this as well. It totally separates your changes from anything originally there.

EDIT: I should have finished reading your entire comment. Yea, this is basically what you said at the end.

```c#
if (_entityWhitelist.IsWhitelistFail(trigger.Whitelist, ent))
    continue;

// DeltaV - TriggerBlacklist
if ((trigger.TriggerBlacklist != null && triggerPool.Any(otherTrigger => trigger.TriggerBlacklist.Contains(otherTrigger.ID)))
  || triggerPool.Any(otherTrigger => otherTrigger.TriggerBlacklist != null && otherTrigger.TriggerBlacklist.Contains(triggerId)))
    continue;
// END DeltaV

```

Reply:
Splitting into separate if probably makes more sense yea